discord_bot/discord_bot.py

```diff
@@ -101,29 +127,33 @@ def get_gateway(self, **kwargs):
                                                                     self.config["gateway_encoding"])) as websocket:
             self.websocket = websocket
             while True:
-                response = await self.websocket.recv()
-                response = json.loads(response)
-                print("{}: {}".format(Opcodes(response["op"]).name, response))
-                if response["op"] == Opcodes.HELLO:
-                    asyncio.ensure_future(self.handshake(response))
-                elif response["op"] == Opcodes.HEARTBEAT_ACK:
+                message = await self.websocket.recv()
+                message = json.loads(message)
+                print("{}: {}".format(Opcodes(message["op"]).name, message))
+                if message["op"] == Opcodes.HELLO:
+                    asyncio.ensure_future(self.handshake(message))
+                elif message["op"] == Opcodes.HEARTBEAT_ACK:
                     pass
-                elif response["op"] == Opcodes.INVALID_SESSION:
+                elif message["op"] == Opcodes.INVALID_SESSION:
+                    self.last_seq = message["s"]
                     print('Invalid Session')
-                elif response["op"] == Opcodes.DISPATCH:
-                    self.last_seq = response["s"]
-                    event = response["t"]
-                    if event == 'READY':
-                        self.session_id = response["d"]["session_id"]
+                elif message["op"] == Opcodes.DISPATCH:
+                    self.last_seq = message["s"]
+                    event = message["t"]
+                    if event == EVENTS[1]:
+                        self.session_id = message["d"]["session_id"]
                 else:
-                    print(response)
+                    print(message)
 
-    async def send_message(self, content):
+    async def send_message(self, recipient_id, content):
         """
         Post a message into the given channel
-        :param content: message sent
-        :return: Fires a 'Message Create' Gateway event
+        :param recipient_id: num - the recipient to open a DM channel with
+        :param content: obj - message sent
+        :return: dict - Fires a 'Message Create' Gateway event
         """
+        channel = requests.post("{}", json={"recipient_id": recipient_id}).json()
```

Comment:
This looks really wrong since we're not posting to a URL, the response should probably be stored in a separate variable to make it more clear what's going on, and eventually we want to get rid of the `requests` library and switch to `aiohttp` (async IO http). For now let's just throw a TODO over this whole method and fix it later since this PR is growing out of control.